### PR TITLE
 Add resolution parameter to OpenTok.start_archive(), (fixes #121)

### LINKED ISSUE #121: OpenTok.start_archive() doesn't have parameter 'resolution'
OpenTok.start_archive() doesn't have parameter 'resolution'. Python SDK is simply through requests.post to use Opentok REST API and it doesn't pass the parameter 'resolution' in the payload json which is default of '640x480'.

### PR DESCRIPTION
**Pull request by hust-whw**

### CODE REVIEW

File: opentok/opentok.py [hunk 1]
```diff
@@ -311,7 +311,7 @@ def archive_url(self, archive_id=None):
             url = url + '/' + archive_id
         return url
 
-    def start_archive(self, session_id, has_audio=True, has_video=True, name=None, output_mode=OutputModes.composed):
+    def start_archive(self, session_id, has_audio=True, has_video=True, name=None, output_mode=OutputModes.composed,resolution="640x480"):
```
Comment (Lucashuang0802) [inline opentok/opentok.py]: Can we use enum or flag instead of a string?

Comment (Lucashuang0802) [inline opentok/opentok.py]: Looks good to me in general. We just need to take care of some corner cases:
```
if resolution != "640x480" and resolution != "1280x720":
  resolution = "640x480"
```

**Comment by maikthomas** [comment]
We should add a test for this.
There's already tests for this method here: https://github.com/opentok/Opentok-Python-SDK/blob/master/tests/test_archive_api.py
so most of the work is already done, it's just adding a new case with the new parameter and checking the resolution I imagine/

File: opentok/opentok.py [hunk 2]
```diff
@@ -359,7 +364,7 @@ def start_archive(self, session_id, has_audio=True, has_video=True, name=None, o
         elif response.status_code == 403:
             raise AuthError()
         elif response.status_code == 400:
-            raise RequestError("Session ID is invalid")
+            raise RequestError("Session ID is invalid or resolution error")
```
Comment (maikthomas) [inline opentok/opentok.py]: I think if possible we need to determine which factor caused this error and raise a specific error.

File: opentok/opentok.py (hunk 1 above)
Comment (maikthomas) [inline opentok/opentok.py]: Below (and from opentok docs) it says 
> If you set this 
          parameter and set the output_mode parameter to OutputModes.individual, the call to the 
          start_archive() method results in an error.`

However, we're setting a default value, and unconditionally adding to the payload.
This sounds like it will break individual mode.

I think maybe the resolution should only be added to payload if individual mode is not selected.

Comment (Lucashuang0802) [reply opentok/opentok.py]: @maikthomas Thanks for bringing this up. If that is the case, we need to take care of this condition carefully.

Comment (Lucashuang0802) [reply opentok/opentok.py]: I agree if the response body contains that property.

File: opentok/opentok.py (hunk 2 above)
Comment (hust-whw) [reply opentok/opentok.py]: Yeah. Cannot agree more. However, I have triggered resolution error (set resolution different string or other type of value), the status_code of response is 400 as well. In addition, the document of OpenAPI also describe the status_code will be 400. I think the python SDK only can determine the error after the OpenAPI respond a different status_code.

**Comment by maikthomas** [comment]
@hust-whw just to let you know that this is in our pipeline, we intend to add the remaining code/tests.
